modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/CommandHandler.java

```diff
@@ -155,14 +160,34 @@ public static IgniteLogger setupJavaLogger(String appName, Class<?> cls) {
      *
      */
     public CommandHandler() {
-        logger = setupJavaLogger("control-utility", CommandHandler.class);
+        this(setupJavaLogger("control-utility", CommandHandler.class));
     }
 
     /**
      * @param logger Logger to use.
      */
     public CommandHandler(IgniteLogger logger) {
         this.logger = logger;
+        Iterable<CommandsProvider> it = U.loadService(CommandsProvider.class);
+
+        cmds = new HashMap<>();
+
+        new CommandsProviderImpl().commands().forEach((k, v) -> cmds.put(k.toLowerCase(), v));
+
+        if (!F.isEmpty(it)) {
+            for (CommandsProvider provider : it) {
```

Comment:
It seems useful to add logging of what exeternal commands were registered.

Reply:
There can be tools which parse the output of control.sh script and new unexpected output can break these tools. At least for "info" logging level. But we can use "debug" logging level.